README.md

```diff
@@ -117,23 +117,25 @@ new GitHubRunners(this, 'runners', {
 Another way to customize runners is by modifying the image used to spin them up. The image contains the [runner][5], any required dependencies, and integration code with the provider. You may choose to customize this image by adding more packages, for example.
 
 ```typescript
-const myBuilder = new CodeBuildImageBuilder(this, 'image builder', {
-  dockerfilePath: FargateRunner.LINUX_X64_DOCKERFILE_PATH,
-  runnerVersion: RunnerVersion.specific('2.291.0'),
-  rebuildInterval: Duration.days(14),    
+const myBuilder = CodeBuildRunnerProvider.imageBuilder(this, 'image builder', {
+   dockerfilePath: FargateRunner.LINUX_X64_DOCKERFILE_PATH,
+   runnerVersion: RunnerVersion.specific('2.291.0'),
+   rebuildInterval: Duration.days(14),
 });
-myBuilder.setBuildArg('EXTRA_PACKAGES', 'nginx xz-utils');
+myBuilder.addComponent(
+  RunnerImageComponent.custom({ commands: ['apt install -y nginx xz-utils'] })
```

Comment:
with this approach, how is order of the commands controlled?
for example, previously, I needed to customize Dockerfile to have additional packages installed, as well as docker login just at the end of the Dockerfile

Reply:
You can override all components if you need to set the order of the basic components too. For example:

```
FargateRunnerProvider.newImage(this, 'builder', {
  components: [
        RunnerImageComponent.requiredPackages(),
        RunnerImageComponent.custom({ name: 'packages', commands: ['apt install -y nginx xz-utils']] }),
        RunnerImageComponent.runnerUser(),
        RunnerImageComponent.git(),
        RunnerImageComponent.githubCli(),
        RunnerImageComponent.awsCli(),
        RunnerImageComponent.custom({ name: 'more-packages', commands: ['apt install -y nginx xz-utils']] }),
        RunnerImageComponent.githubRunner(props?.runnerVersion ?? RunnerVersion.latest()),
  ],
});
```

Though I'm not sure why you'd need to set the order in this example. Wouldn't it be enough to add the packages and then docker login command? I am not sure I understand why they need to happen before the standard components.

Reply:
it's not before but after that is critical, because it's a docker login command
so when you add custom components, it's a guarantee that they come after the default ones?

Reply:
Right now they will be executed after the default components, yeah. You are talking about `docker login` inside the built image, right? Or are you talking about login to pull the base image?

Reply:
Ok great, then it should work.
Yes, docker login inside the built image, this is to avoid rate limiting from docker repositories